Does finally block get executed if we insert a return statement inside try block of a try-catch-finally?

Yes.

Note there are some cases in which the finally block will not get executed
if the virtual machine exits during try/catch
if the thread which is executing the try/catch